RFS/src/main/java/com/rfs/common/OpenSearchClient.java

```diff
@@ -118,8 +118,26 @@ private Optional<ObjectNode> createObjectIdempotent(
             .retryWhen(Retry.backoff(3, Duration.ofSeconds(1)).maxBackoff(Duration.ofSeconds(10)))
             .block();
 
-        if (response.code == HttpURLConnection.HTTP_NOT_FOUND) {
-            client.put(objectPath, settings.toString(), context.createCheckRequestContext());
+        if (getResponse.code == HttpURLConnection.HTTP_NOT_FOUND) {
```

Comment:
This line made me do a double take, what do you think about making a bool local called 'objectDoesNotExist` that gets the value of `getResponse.code == HttpURLConnection.HTTP_NOT_FOUND` so its clear what the body of this if block is doing?

Reply:
Also, inverting the condition to return early if the object exists would help make this clearer yet.

Reply:
Will queue the change for the next PR.